src/arc.rs

```diff
@@ -65,7 +65,15 @@ impl PathArc {
     ///
     /// [0]: https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.join
     pub fn join<P: AsRef<Path>>(&self, path: P) -> PathArc {
-        PathArc::from(self.0.join(path))
+        // Clone the current path
+        let mut joined_path = self.as_path().to_path_buf();
+        // Append the given path component by component so that path separator are correctly
```

Comment:
Please change comment to something like:

```
// issue #34: windows preserves '\\' characters with `Path.push`.
```

Reply:
What do you think of `// Issue #34: windows automatically convert '/' into '\\' with Path.push`? It seems clearer for me.

Reply:
Ya, something like that sounds better. Maybe:

```
// Issue #34: convert '/' into '\\' for windows
```